File: ios/Animation/GroupAnimation.m
//
//  GroupAnimation.m
//  splashmoduleios
//
//  Created by Smith Charles on 22/09/2020.
//

#import "GroupAnimation.h"
#import "AnimationsList.h"
#import "Splash.h"

@implementation GroupAnimation

-(instancetype)init:(int)priority{
    _priority = priority;
  return self;
}

-(void)addAnimation:(ObjectAnimation*)object{
    [object setPriority:_priority];
    [AnimationsList appendInAnimationList:object];
}

@end

File: android/src/main/java/com/blitzapp/animatedsplash/animation/GroupAnimation.java
package com.blitzapp.animatedsplash.animation;


import static com.blitzapp.animatedsplash.animation.Constants.GROUP;
import static com.blitzapp.animatedsplash.animation.Splash.animatedObjectList;

import com.blitzapp.animatedsplash.AnimationsList;

import java.util.ArrayList;
import java.util.List;

public class GroupAnimation {

    public String type = GROUP;
    public int priority = 0;
    private List<ObjectAnimation> group = new ArrayList<>();

    public GroupAnimation(int priority) {
        this.priority = priority;
    }

    public void addAnimation(ObjectAnimation object){
        object.priority = priority;
        AnimationsList.appendInAnimationList(object);
    }

  }